test/helpers/ArtifactsReader.sol

```diff
@@ -20,4 +20,17 @@ abstract contract ArtifactsReader is Script {
             return address(0);
         }
     }
+
+    function _getOtherChainDeployment(string memory contractName, uint256 chainId) internal virtual returns (address) {
```

Comment:
```suggestion
    function _getChainDeployment(string memory contractName, uint256 chainId) internal virtual returns (address) {
```

you can just keep the same method name as the one above and make the one above to call this one, would look something like this:

```
function _getAddressesFile() internal view virtual returns (string memory) {
        return _getAddressesFile(block.chainid);
    }

    function _getAddressesFile(uint256 chainId) internal view virtual returns (string memory) {
        string memory root = vm.projectRoot();
        return string.concat(root, "/test/artifacts/", vm.toString(chainId), "/addresses.json");
    }

    function _getChainDeployment(string memory contractName) internal virtual returns (address) {
        return _getChainDeployment(contractName, block.chainid);
    }

    function _getChainDeployment(string memory contractName, uint256 chainId) internal virtual returns (address) {
        try vm.readFile(_getAddressesFile(chainId)) returns (string memory json) {
            try vm.parseJsonAddress(json, string.concat(".", contractName)) returns (address addr) {
                return addr;
            } catch {
                return address(0);
            }
        } catch {
            return address(0);
        }
    }
    ```

Reply:
done